### PR TITLE
hal: gigadevice: Update gd32f4xx to firmware 3.0.0

### PR DESCRIPTION
**Pull request by duchesne9al**
This PR is associated with  : 
- [hal_gigadevice/pull/24](https://github.com/zephyrproject-rtos/hal_gigadevice/pull/24)
- [hal_gigadevice/pull/25](https://github.com/zephyrproject-rtos/hal_gigadevice/pull/25)

Signed-off-by: Alexandre Duchesne [alexandre.duchesne@rtone.fr](mailto:alexandre.duchesne@rtone.fr)

### CODE REVIEW

**Review by gmarull — CHANGES_REQUESTED** [review]
HAL patches are not complete, see for example https://github.com/zephyrproject-rtos/hal_gigadevice/pull/22/commits/48a1b6782d330d0ea6163717b1cfcde877e6440d, also please use `pull/24/head` in the manifest until it gets merged

**Comment by gmarull** [comment]
@duchesne9al you can update sha in west.yaml, hal PR has been merged

**Comment by duchesne9al** [comment]
> @duchesne9al you can update sha in west.yaml, hal PR has been merged

It's done.
Thanks for the review :)

**Review by gmarull — DISMISSED** [review]

File: drivers/pinctrl/pinctrl_gd32_af.c [hunk 1]
```diff
@@ -25,7 +25,7 @@ BUILD_ASSERT((GD32_OSPEED_2MHZ == GPIO_OSPEED_2MHZ) &&
 #else
 	     (GD32_OSPEED_25MHZ == GPIO_OSPEED_25MHZ) &&
 	     (GD32_OSPEED_50MHZ == GPIO_OSPEED_50MHZ) &&
-	     (GD32_OSPEED_200MHZ == GPIO_OSPEED_200MHZ) &&
+	     (GD32_OSPEED_200MHZ == GPIO_OSPEED_MAX) &&
```
Comment (nandojve) [inline drivers/pinctrl/pinctrl_gd32_af.c:28]: I think we should update at Zephyr side from GD32_OSPEED_200MHZ to GD32_OSPEED_MAX too. The main reason is that there is no more mention in hal_gigadevice and keep as is will create an inconsistence at Zephyr side.

**Review by nandojve — CHANGES_REQUESTED** [review]

File: drivers/pinctrl/pinctrl_gd32_af.c (hunk 1 above)
Comment (duchesne9al) [reply drivers/pinctrl/pinctrl_gd32_af.c]: Yes indeed.

Comment (gmarull) [reply drivers/pinctrl/pinctrl_gd32_af.c]: good point

**Review by gmarull — APPROVED** [review]

**Review by nandojve — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**